### PR TITLE
Add keyboard shortcut: "V" to toggle between TextViews

### PR DESCRIPTION
**Pull request by khemarato**
This PR (built on top of https://github.com/suttacentral/suttacentral/pull/3236 so please ignore the extra string changes) takes a first stab at one of the keyboard shortcuts in https://github.com/suttacentral/suttacentral/issues/2308

### CODE REVIEW

File: client/elements/addons/sc-top-sheet-views.js
Comment (khemarato) [inline client/elements/addons/sc-top-sheet-views.js:187]: This is what this looks like in both light and dark themes:

Screenshot from 2024-08-04 14-29-08

Screenshot from 2024-08-04 14-29-18

Happy to change this styling to conform to whatever style guidelines you have.

File: client/elements/addons/sc-top-sheet-views.js
Comment (khemarato) [inline client/elements/addons/sc-top-sheet-views.js]: This timeout is absolutely necessary to make it work, but I can't for the life of me figure out why.  Without this `show/hide`, the toast pops up showing that the key press was registered, but the layout doesn't actually change.  Any ideas, @ihongda ?

File: client/elements/addons/sc-top-sheet-views.js
Comment (khemarato) [inline client/elements/addons/sc-top-sheet-views.js:345]: I was expecting this callback to be called when navigating from a sutta page to e.g. the homepage, but it wasn't called.  It seems that the top sheets are never unloaded?

File: client/elements/menus/sc-action-items.js
Comment (khemarato) [inline client/elements/menus/sc-action-items.js:312]: While I'm here, I also added a keyboard shortcut to close all the top sheets: `esc`

File: client/localization/elements/viewoption_en.json
Comment (khemarato) [inline client/localization/elements/viewoption_en.json]: The other string changes were pulled in automatically by the build (from where?) except for this one which I changed to get the helptext updated.  What is the correct way to update this string, @sujato ?

**Review by khemarato — COMMENTED** [review]
cc @thesunshade

**Comment by thesunshade** [comment]
Thanks!!!!!

I am able to break it, though. 

1. Go to an alligned sutta, e.g. MN 5.
2. Cycle through the views by pressing `v`.
3. Click on the parent vagga in the breadcrumb trail
image
4. click on a new aligned translation, e.g. MN 6
5. Cycle through the views by pressing `v`.

**What doesn't happen:**
* the actual view does not change (meaning that the text itself does not change)

**What does happen**
* the snack bar announces the change
* the params in the url change

**What probably happens**
* the radio buttons change

HOWEVER… once you click on the "views" menu, then the change is applied and the new view is shown. After that, the `v` toggle works just fine.

Note, the `esc` key seems to work immediately, which probably makes sense. Opening the "info" panel doesn't get the view toggling working, which probably makes sense.

File: client/elements/addons/sc-top-sheet-views.js
Comment (khemarato) [reply client/elements/addons/sc-top-sheet-views.js]: Thanks for playing with it @thesunshade ! Your bug is certainly the same one this is attempting to work around. It will definitely need a real fix before this can merge. That's waiting on @ihongda's knowledge: what gets hooked up only after the top panel shows on a given sutta?

File: client/localization/elements/viewoption_en.json
Comment (thesunshade) [reply client/localization/elements/viewoption_en.json]: I think this has to be updated in `bilara-data` and then go through all its magic.
I'm wondering if this help text should get its own new segment.

File: client/elements/addons/sc-top-sheet-views.js
Comment (ihongda) [reply client/elements/addons/sc-top-sheet-views.js]: > This timeout is absolutely necessary to make it work, but I can't for the life of me figure out why. Without this `show/hide`, the toast pops up showing that the key press was registered, but the layout doesn't actually change. Any ideas, @ihongda ?

Bhante 🙏🏼 

This part of the code can be placed in the `firstUpdated` event of lit.
>Called after the component's DOM has been updated the first time, immediately before [updated()](https://lit.dev/docs/components/lifecycle/#updated) is called.

In the firstUpdated event, we should be able to remove the `setTimeout`.

File: client/elements/addons/sc-top-sheet-views.js
Comment (ihongda) [inline client/elements/addons/sc-top-sheet-views.js]: Using `const` or `let` instead of `var` is more in line with the specification 🙏🏼

File: client/elements/addons/sc-top-sheet-views.js
Comment (ihongda) [reply client/elements/addons/sc-top-sheet-views.js:345]: Yes, this component is not actively uninstalled from the DOM.

File: client/localization/elements/viewoption_en.json
Comment (khemarato) [reply client/localization/elements/viewoption_en.json]: @thesunshade - Can you take the lead on this upstream change? :pray:

Comment (thesunshade) [reply client/localization/elements/viewoption_en.json]: Sure. Just give me a poke when the shortcut is live if I don't notice.

File: client/elements/addons/sc-top-sheet-views.js
Comment (khemarato) [reply client/elements/addons/sc-top-sheet-views.js]: What are you talking about @ihongda? You mean put the addEventListener to the firstUpdated callback instead of the connectedCallback?  Cause that doesn't work.  I'm telling you that there's something that's only getting hooked up after the top sheet actually _opens_ (firstUpdated is called before that).  There's some function I need to call here (or some other logic somewhere else that I need to change?) to make sure that `actions.chooseSegmentedSuttaTextView` actually does what it's supposed to.

Comment (khemarato) [reply client/elements/addons/sc-top-sheet-views.js]: @ihongda - Okay. I figured it out. `sc-text-bilara.shouldRestoreUserSettings` is `false` by default and is suppressing the user-selected setting until the user opens the control panel, I assume to allow the url params to take effect without changing the user's settings.  In this case, the correct thing to do (?) is to set shouldRestoreUserSettings to `true` after the user hits one of the keyboard shortcuts.  Will update the PR with that logic.

File: client/elements/addons/sc-top-sheet-views.js
Comment (khemarato) [inline client/elements/addons/sc-top-sheet-views.js:358]: @ihongda - this is the magic function call I needed to make it work.

Comment (ihongda) [reply client/elements/addons/sc-top-sheet-views.js:358]: 👍🏼

**Review by ihongda — APPROVED** [review]
Thanks, Bhante 🙏🏼

**Comment by thesunshade** [comment]
My suggestion would be to hold off on the interface changes for now. There are some folks who will start localizing things immediately and it's probably good to have further discussions on the best way to notify users of the hot-keys.

**Comment by khemarato** [comment]
The string change here will get overwritten on the next `make` right?

**Comment by thesunshade** [comment]
> The string change here will get overwritten on the next `make` right?

Yes, sorry. I was speaking more towards HongDa. I do believe that your change will be lost.

It's a discussion to be held probably in the issue, but I'm not even sure what Bhante had in mind. There will be a few that won't have an easy place to note in the existing interface.

BTW, I forgot to test what happens when you type a `v` when entering something in the search field. Did you?

**Comment by khemarato** [comment]
Thanks @thesunshade - That was indeed getting captured. Somehow I thought it would automatically ignore `input` keypresses.  Anyway, that's now fixed and I've removed the string change. :)

**Pull request merged**
